nservicebus/best-practices.md

```diff
@@ -0,0 +1,130 @@
+
+For a high-level overview of infrastructure concerns and behaviors, see the blog post [Infrastructure soup](https://particular.net/blog/infrastructure-soup).
+
+:x: **DO NOT catch exceptions in a message handler**
+
+NServiceBus relies on a message handler completing _without_ throwing an exception to know that the message was processed successfully. If an exception is thrown, the message can be retried until it succeeds, which resolves many transient errors. (See the blog post [I caught an exception. Now what?](https://particular.net/blog/but-all-my-errors-are-severe).) If too many retries occur, the message is moved to an error queue where a developer can figure out what's wrong with it and correct the underlying issue.
+
+If exceptions are caught in a message handler, the message will be consumed, without the ability to be retried, regardless of the error.
+
+If it is necessary to catch certain kinds of exceptions, consider throwing a new exception with a more specific message to give context about what is going on in the message handler that led to the failure. Always be sure to include the original error in the constructor of the new exception, so that the `InnerException` property will show the stack trace of the original failure.
+
+
+:x: **DO NOT [enable installers](/nservicebus/operations/installers.md) in production**
+
+In a development environment, installers create dependencies that the endpoint needs to run, such as queues in the message transport and tables or schema in the persistence store. This is meant to make the development process fast and friction-free because changes are occurring at a rapid pace.
+
+In production scenarios, different permissions are needed to administer resources such as database tables and queues, and the endpoint should not run in production with permissions to alter these resources.
+
+In production, the creation of queues, database tables, and other tables should be scripted as a part of a DevOps process when the endpoint is installed. Methods to administer these resources differ for every message transport and persistence package. For many packages, Particular will provide a `dotnet` global tool, PowerShell library, or other means of performing operational scripting to create endpoints.
+
+
+:heavy_check_mark: **DO use the [Outbox feature](/nservicebus/outbox/) to provide consistency between business data modifications and messaging operations**
+
+Because message queues generally do not support any form of transactions, message handlers that modify business data in a database run into problems when the messages that are sent become inconsistent with the changes made to business data, resulting in ghost messages or phantom records.
+
+Using the Outbox guarantees exactly-once message processing by piggybacking on the local database transaction used to store business data.
```

Comment:
Not sure if I'd use 'piggybacking'. I'm not sure I even knew what that word meant, until I met you! :)